### PR TITLE
[KernelGen][Nvidia] Add _softmax operator with Triton kernel

### PR DESCRIPTION
**Pull request by CoyeCHEN**
## Summary

Registers the canonical `_softmax` entry through a thin wrapper that calls the existing FlagGems implementation. No new kernel is introduced and no source implementation line is changed.

Target registrations added or remapped: `default`.

## Overload Coverage

| ATen overload | status | FlagGems wrapper | evidence |
|---|---|---|---|
| `default` | `remap` | `_softmax` | GEMS _SOFTMAX |

## Testing

- Direct ATen dispatch tests: 48 passed, 0 failed, 0 skipped.
- Complete changed test file on GPU: 216 passed, 0 skipped, 0 failed.
- Target ATen tests in CPU quick-reference mode: 2 passed, 0 failed, 0 skipped.
- Complete changed test file in CPU quick-reference mode: 10 passed, 0 skipped, 0 failed.
- Complete core benchmark file: 5 passed, 0 skipped, 0 failed.
- Base: `2e3bd16e580e0a2cd3e74cb9b0c4d28f5bc03e12` (`flagos-ai/FlagGems:master`).

### `_softmax`

| dtype | shape | Torch (ms) | FlagGems (ms) | speedup | TFLOPS |
|---|---:|---:|---:|---:|---:|
| float16 | `1073741824` | 261.621918 | 283.554413 | 0.923 | 0.000 |
| float16 | `64, 64` | 0.006336 | 0.006080 | 1.042 | 0.000 |
| float16 | `4096, 4096` | 0.094304 | 0.036416 | 2.590 | 0.000 |
| float16 | `64, 512, 512` | 0.500848 | 0.071968 | 6.959 | 0.000 |
| float16 | `1024, 1024, 1024` | 16.560385 | 2.744288 | 6.034 | 0.000 |
| float32 | `1073741824` | 377.847900 | 296.422241 | 1.275 | 0.000 |
| float32 | `64, 64` | 0.006368 | 0.006048 | 1.053 | 0.000 |
| float32 | `4096, 4096` | 0.093152 | 0.046592 | 1.999 | 0.000 |
| float32 | `64, 512, 512` | 0.625664 | 0.087040 | 7.188 | 0.000 |
| float32 | `1024, 1024, 1024` | 17.535168 | 3.764480 | 4.658 | 0.000 |
| bfloat16 | `1073741824` | 265.953522 | 298.669647 | 0.890 | 0.000 |
| bfloat16 | `64, 64` | 0.006336 | 0.006080 | 1.042 | 0.000 |
| bfloat16 | `4096, 4096` | 0.093760 | 0.036544 | 2.566 | 0.000 |
| bfloat16 | `64, 512, 512` | 0.502208 | 0.079328 | 6.331 | 0.000 |
| bfloat16 | `1024, 1024, 1024` | 16.555392 | 2.852672 | 5.803 | 0.000 |

Complete result count: 15; geometric mean speedup: **2.479x**.

## Multi-backend Testing

| Backend | Accuracy | Benchmark | Notes |
|---|---|---|---|
| Nvidia H20 | PASS (48 direct / 216 full-file GPU / 10 quick-CPU tests) | PASS (15 complete target rows) | Primary validation |
| Other backends | N/A | N/A | Not run in this local pipeline |

## Files Changed

- `benchmark/test_softmax.py`: +17 / -0
- `conf/operators.yaml`: +11 / -2
- `src/flag_gems/__init__.py`: +1 / -1
- `src/flag_gems/ops/__init__.py`: +2 / -0
- `src/flag_gems/ops/_softmax.py`: +26 / -0
- `tests/test_softmax.py`: +20 / -0

### CODE REVIEW

File: src/flag_gems/__init__.py [hunk 1]
```diff
@@ -153,7 +153,7 @@ def torch_ge(v):
     ("_scaled_mm.out", scaled_mm_out, lambda: torch_ge("2.5")),
     ("_segment_reduce_backward", _segment_reduce_backward),
     ("_segment_reduce_backward.out", _segment_reduce_backward_out),
-    ("_softmax", softmax),
+    ("_softmax", _softmax),
```
Comment (Caeruleann) [inline src/flag_gems/__init__.py:163]: Don't delete softmax.

File: src/flag_gems/__init__.py [hunk 2]
```diff
@@ -160,7 +160,7 @@ def torch_ge(v):
...
```
Comment (CoyeCHEN) [inline src/flag_gems/__init__.py:163]: Thanks for the review! I'd like to clarify that `softmax` is **not deleted**:

1. The change `("_softmax", softmax)` → `("_softmax", _softmax)` only remaps the `_softmax.default` ATen key registration to a new canonical wrapper
2. The `softmax` function itself remains fully functional and is still exported in `ops/__init__.py:1448`
3. The new `_softmax` wrapper simply calls the existing `softmax()` implementation - no code or functionality is lost

**This follows the exact same pattern as the merged PR #5180** (`native_layer_norm`):
- After #5180 merged, master now has `("native_layer_norm", native_layer_norm)`
- The original `layer_norm` function also has zero direct ATen registrations in master's `_FULL_CONFIG`
- But `layer_norm` remains exported and is called by the new wrapper

This is the standard approach for alias registration in this codebase.

File: benchmark/test_softmax.py [hunk 3]
```diff
@@ -20,6 +20,23 @@
 from . import base, consts
 
 
+def _softmax_input_fn(shape, dtype, device):
+    inp = base.generate_tensor_input(shape, dtype, device)
+    dim = 1 if inp.ndim > 1 else 0
+    yield inp, dim, False
+
+
+@pytest.mark.underscore_softmax
```
Comment (Caeruleann) [inline benchmark/test_softmax.py:29]:
```suggestion
@pytest.mark.softmax
```

File: conf/operators.yaml [hunk 4]
```diff
@@ -9051,10 +9051,19 @@ ops:
       - Math
     stages:
       - alpha: '5.4'
-  - id: softmax
-    description: Apply a softmax function.
+  - id: _softmax
```
Comment (Caeruleann) [inline conf/operators.yaml:9054]:
```suggestion
  - id: softmax
```

File: src/flag_gems/__init__.py (hunk 2 above)
Comment (Caeruleann) [inline src/flag_gems/__init__.py:163]:
```suggestion
    ("_softmax", softmax),
    ("_softmax", _softmax),
```

**Comment by CoyeCHEN** [comment]
Closing this PR as `_softmax` is already covered by the merged operator `softmax` (#4835).

**Why the duplicate exists:**
- ATen has both `aten::softmax.int` and `aten::_softmax` — the latter is an internal variant
- PyTorch routes both to the same implementation path
- FlagGems' registration of `softmax` already intercepts both schemas

**Verification (on current master 0a54e6da7):**
```python
import torch
import flag_gems

x = torch.randn(16, 128, device='cuda', requires_grad=True)

with flag_gems.use_gems():
    # Both paths reach the same FlagGems kernel
    out1 = torch.softmax(x, dim=-1)
    out2 = torch._softmax(x, dim=-1, half_to_float=False)
    
print(torch.allclose(out1, out2))  # True
```

The `_softmax` entry point does not need separate registration or testing — the existing `softmax` implementation handles it.
